src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1065,923 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+                    except NoTTYException:
+                        pass
+                    if overwrite:
+                        existing[key].remove(j)
+                    else:
+                        msg = 'A different object named {} already exists in {} in your kubeconfig file.'
+                        telemetry.set_exception(exception='A different object with same name exists in the kubeconfig file', fault_type=consts.Different_Object_With_Same_Name_Fault_Type,
+                                                summary=msg.format(i['name'], key))
+                        raise CLIError(msg.format(i['name'], key))
+        existing[key].append(i)
+
+
+def _resolve_service_principal(client, identifier):  # Uses service principal graph client
+    # todo: confirm with graph team that a service principal name must be unique
+    result = list(client.list(filter="servicePrincipalNames/any(c:c eq '{}')".format(identifier)))
+    if result:
+        return result[0].object_id
+    if utils.is_guid(identifier):
+        return identifier  # assume an object id
+    error = CLIError("Service principal '{}' doesn't exist".format(identifier))
+    error.status_code = 404  # Make sure CLI returns 3
+    raise error
+
+
+def client_side_proxy_wrapper(cmd,
+                              client,
+                              resource_group_name,
+                              cluster_name,
+                              token=None,
+                              path=os.path.join(os.path.expanduser('~'), '.kube', 'config'),
+                              overwrite_existing=False,
+                              context_name=None,
+                              api_server_port=consts.API_SERVER_PORT,
+                              client_proxy_port=consts.CLIENT_PROXY_PORT):
+
+    cloud = send_cloud_telemetry(cmd)
+    args = []
+    operating_system = platform.system()
+    proc_name = f'arcProxy{operating_system}'
+
+    if(check_process(proc_name)):
+        raise CLIError('Another instance of proxy already running')
+
+    port_error_string = ""
+    if check_if_port_is_open(api_server_port):
+        port_error_string += f'Port {api_server_port} is already in use. Please select a different port with --api-server option.\n'
+    if check_if_port_is_open(client_proxy_port):
+        port_error_string += f'Port {client_proxy_port} is already in use. Please select a different port with --client-proxy-option.\n'
+    if port_error_string != "":
+        raise CLIError(port_error_string)
+
+    signal.signal(signal.SIGINT, ctrlc_handler)
+    # Creating installation location, request uri and older version exe location depending on OS
+    if(operating_system == 'Windows'):
+        install_location_string = f'.clientproxy\\arcProxy{operating_system}{consts.CLIENT_PROXY_VERSION}.exe'
+        requestUri = f'https://k8sconnectcsp.blob.core.windows.net/release12-03-21/arcProxy{operating_system}{consts.CLIENT_PROXY_VERSION}.exe'
```

Comment:
>k8sconnectcsp.blob.core.windows.net [](start = 31, length = 35)

Nit: Suggest to declare as constant

Reply:
URL, release version, etc as contant and then construct the url

---
In reply to: [596737261](https://github.com/Azure/azure-cli-extensions/pull/3001#discussion_r596737261) [](ancestors = 596737261)

Reply:
@atharvamulmuley , added a URL section in constants.py, lets add urls there

Reply:
yep, added those params in constants and reconstructing the uri from those.